src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MetadataHelper.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace System.Reflection.Emit
+{
+    // This static helper class adds common entities to a Metadata Builder.
+    internal static class MetadataHelper
+    {
+        internal static AssemblyReferenceHandle AddAssemblyReference(Assembly assembly, MetadataBuilder metadata)
+        {
+            AssemblyName assemblyName = assembly.GetName();
+
+            return AddAssemblyReference(metadata, assemblyName.Name!, assemblyName.Version, assemblyName.CultureName, assemblyName.GetPublicKey(), (AssemblyFlags)assemblyName.Flags);
+        }
+
+        internal static AssemblyReferenceHandle AddAssemblyReference(MetadataBuilder metadata, string name, Version? version, string? culture, byte[]? publicKey, AssemblyFlags flags)
+        {
+            return metadata.AddAssemblyReference(
+                name: metadata.GetOrAddString(name),
+                version: version ?? new Version(0, 0, 0, 0),
+                culture: (culture == null) ? default : metadata.GetOrAddString(value: culture),
+                publicKeyOrToken: (publicKey == null) ? default : metadata.GetOrAddBlob(publicKey),
+                flags: flags,
+                hashValue: default); // not sure where to find hashValue.
```

Comment:
> // not sure where to find hashValue.

I believe this is to verify `.file` directive assemblies. You can search for "assemblyRefDecl" in the runtime. Since we don't support creating these types of assemblies with AssemblyBuilder (or referencing them), we shouldn't have to do anything special to support this.

From ECMA335 (see `.hash` below):
```
 II.6.3 Referencing assemblies:
...
AsmRefDecl ::= Description Clause
| .hash ‘=’ ‘(’ Bytes ‘)’ Hash of referenced assembly §II.6.2.3
| .custom CustomDecl Custom attributes §II.21
| .culture QSTRING Culture of the referenced assembly §II.6.2.1.2
| .publickeytoken ‘=’ ‘(’ Bytes ‘)’ The low 8 bytes of the SHA-1 hash of the 
originator's public key.
§II.6.3
| .publickey ‘=’ ‘(’ Bytes ‘)’ The originator’s full public key §II.6.2.1.3
| .ver Int32 ‘:’ Int32 ‘:’ Int32 ‘:’ Int32 Major version, minor version, build, and 
revision
§II.6.2.1.4
```

and 
```
II.6.2.3 Associating files with an assembly 
...
Assemblies can be associated with other files (such as documentation and other files that are used 
during execution). The declaration .file is used to add a reference to such a file to the manifest of the 
assembly: (See §II.22.19) 
Decl ::= Clause
 .file [ nometadata ] Filename .hash ‘=’ ‘(’ Bytes ‘)’ [ .entrypoint ]
| … §II.5.10
The attribute nometadata is specified if the file is not a module according to this specification. Files 
that are marked as nometadata can have any format; they are considered pure data files. 
The Bytes after the .hash specify a hash value computed for the file. The VES shall recompute this 
hash value prior to accessing this file and if the two do not match, the behavior is unspecified. The 
algorithm used to calculate this hash value is specified with .hash algorithm (§II.6.2.1.1).
```

where the hash algorithm is assumed to be SHA-1.

Reply:
> I believe this is to verify .file directive assemblies. You can search for "assemblyRefDecl" in the runtime. Since we don't support referencing these types of assemblies with AssemblyBuilder we shouldn't have to do anything special to support this.

Thanks, will update the comment